### ISSUE TITLE
File path too long

### ISSUE THREAD

**Issue by polackgod** [aws/aws-codebuild-docker-images#244]
Attempting to use CodeBuild to compile .NET 4.6.2 web application.  I'm using AWS image: aws/codebuild/windows-base:2.0

My buildspec.yml file is borrowed from https://docs.aws.amazon.com/codebuild/latest/userguide/sample-windows.html where I just changed solution name.
My solution has several NuGet packages that get downloaded.  I see them being downloaded in the log.  However, after a while I get this error:

`
...
Added package 'Microsoft.CodeDom.Providers.DotNetCompilerPlatform.2.0.1' to folder 'C:\codebuild\tmp\output\src863383187\src\git-codecommit.us-east-2.amazonaws.com\v1\repos\MySolution\packages' 
Adding package 'Microsoft.CodeDom.Providers.DotNetCompilerPlatform.2.0.0' to folder 'C:\codebuild\tmp\output\src863383187\src\git-codecommit.us-east-2.amazonaws.com\v1\repos\MySolution\packages' 
Added package 'Microsoft.CodeDom.Providers.DotNetCompilerPlatform.2.0.0' to folder 'C:\codebuild\tmp\output\src863383187\src\git-codecommit.us-east-2.amazonaws.com\v1\repos\MySolution\packages' 
 
Errors in packages.config projects 
    The specified path, file name, or both are too long. The fully qualified file name must be less than 260 characters, and the directory name must be less than 248 characters. 
 
NuGet Config files used: 
    C:\Users\ContainerAdministrator\AppData\Roaming\NuGet\NuGet.Config 
 
Feeds used: 
    https://api.nuget.org/v3/index.json 
 
Installed: 
    67 package(s) to packages.config projects 
 
[Container] 2019/08/19 14:59:06 Command did not exit successfully & "C:\ProgramData\chocolatey\bin\NuGet.exe" restore $env:SOLUTION -PackagesDirectory $env:PACKAGE_DIRECTORY exit status 1 
[Container] 2019/08/19 14:59:09 Phase complete: BUILD State: FAILED 
[Container] 2019/08/19 14:59:09 Phase context status code: COMMAND_EXECUTION_ERROR Message: Error while executing command: & "C:\ProgramData\chocolatey\bin\NuGet.exe" restore $env:SOLUTION -PackagesDirectory $env:PACKAGE_DIRECTORY. Reason: exit status 1 
[Container] 2019/08/19 14:59:09 Entering phase POST_BUILD 
...
`
Seems like the file path is hitting the 260 limit.  Some of the packages are pretty wordy so I guess I'm not surprised.  
Any chance long file path option can be enabled in the image?

Thanks.

**Comment by polackgod** [comment]
FYI...I found a workaround for this problem.
I tried to use powershell to change the registry key to allow long paths but that didn't get me very far.  
Instead I decided to copy the downloaded source code from the standard location (CODEBUILD_SRC_DIR) to something with shorter path (c:\codebuild\src) and then I was able to build my application successfully.  
In addition I had to specify base_directory in the artifact section to be the location I copied source into, otherwise, artifact was created using CODEBUILD_SRC_DIR location which didn't have any of the compiled code.
It all works fine but it's a bit too much gymnastics and results in a brittle solution IMHO.  Having the long paths enabled would be cleaner.
Thanks.

**Comment by TimDrayk** [comment]
Can you please try LongPathTool? I think it can help you to solve this issue.

**Comment by subinataws** [comment]
@polackgod - Did @TimDrayk suggestion help?

**Comment by piotrS-bright** [comment]
@subinataws 
Ended up not using LongPathTool and going with the solution I described in my initial comment: copy downloaded source to a shorter path location.
I couldn't think of a way how LongPathTool could help me better.  I could have used it to copy downloaded source:
`LongPathTool.exe /input $env:CODEBUILD_SRC_DIR /output c:\\shortPath /action copy`
but xcopy works just fine and it's one less tool to include with the build process.

I guess I would need to use LongPathTool when downloaded source path would end up being too long right of the bat but that doesn't occur until I start restoring NuGet packages.  It's possible that some files could be named very "wordy" that would take them past the limit and then I'd need to use LongPathTool but I decided to cross that bridge when I get to it.

Thanks for suggestion.
